### PR TITLE
Performance updates for application start

### PR DESCRIPTION
**Pull request by jhanders34**
- Most of these are minor changes, but are on main line paths that can have cumulative benefits.
- Add isDebugEnabled / isLoggable checks around trace calls.
 - Update to use StringBuilder more efficiently by setting size and reusing an explicit StringBuilder instance instead of the compiler making new ones each time.
- Switch to use String.replace(char,char) instead of replace(String,String) for single character strings.
- Do not eagerly create and store hashText field in UtilsImpl Map objects and just compute it each time since it is only used for trace.
- Update to only use add instead of contains and add when seeing if something can be added to a Set.
- Do not get substring(1) in ZipFileContainer for no reason since nestedContainerEntries is only used in that method.
- ManifestClassPathUtils methods take an Entry object that is adapted from a Container object.  Before this PR the Entry was adapted back to a Container object just to call getEntry on it.  Instead of making a whole new Container object, pass in the original Container that was adapted to the Entry.  Had to put it in a new class since ManifestClassPathUtils is an SPI.

### CODE REVIEW

**Comment by jhanders34** [comment]
#build

**Comment by fritze2** [comment]
Hi Jared, just submitting another test pipeline for comparison to the RTC builds. !pbbeta

**Comment by jhanders34** [comment]
#build

**Comment by jhanders34** [comment]
#libby #build

File: dev/com.ibm.ws.artifact.url/src/com/ibm/ws/artifact/url/internal/WSJarURLStreamHandler.java
Comment (tbitonti) [inline dev/com.ibm.ws.artifact.url/src/com/ibm/ws/artifact/url/internal/WSJarURLStreamHandler.java:1]: Curious .... does this mean that the old code was bypassing the zip handle caching layer?  This seems a potentially significant change.

Comment (jhanders34) [reply dev/com.ibm.ws.artifact.url/src/com/ibm/ws/artifact/url/internal/WSJarURLStreamHandler.java:1]: Yes it was bypassing the caching layer of entries in the zip file.  Of course today we only cache .class files and manifest files.  We may change that to include other files in the near future for other issues.

**Review by tbitonti — APPROVED** [review]
Just one comment on WSJarURLStreamHandler (see the comments).

**Pull request merged**